aiomusiccast/pyamaha.py

```diff
@@ -91,12 +91,19 @@ def connection_made(self, transport):
         self.transport = transport
 
     def datagram_received(self, data, addr):
-        message = data.decode()
+        message_data = None
+        message_str = ""
         try:
-            data = json.loads(message)
-            asyncio.create_task(self.handle_event(data))
+            message_str = data.decode()
+            message_data = json.loads(message_str)
+        except UnicodeDecodeError:
+            _LOGGER.error("Received non UTF-8 compliant message: %s", data)
         except ValueError:
-            _LOGGER.error("Received invalid message: %s", message)
+            _LOGGER.error("Received invalid message: %s", message_str)
+        except Exception:
+            _LOGGER.exception("An unexpected error occurred while handling an UDP message.")
+        finally:
+            asyncio.create_task(self.handle_event(message_data))
```

Comment:
```suggestion
        finally:
            asyncio.create_task(self.handle_event(None))
```

I would explicitly write None here to make it clear that handle_event will be called with None in case of an exception

Reply:
In that case we would always make the call with None as message, wouldn't we? 
We would have to put this into all the except blocks with None and with the message at the end of the try-block, I believe. That was the reason, for that it was implemented this way... If you something else in mind to enhance the readability, feel free 🙂